crates/js_api/src/gui/validation.rs

```diff
@@ -0,0 +1,1056 @@
+use alloy::primitives::{
+    utils::{parse_units, ParseUnits},
+    U256,
+};
+use rain_orderbook_app_settings::gui::{DepositValidationCfg, FieldValueValidationCfg};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum GuiValidationError {
+    #[error("Invalid number for {name}: {value}")]
+    InvalidNumber { name: String, value: String },
+
+    #[error("{name} value {value} is less than minimum {minimum}")]
+    BelowMinimum {
+        name: String,
+        value: String,
+        minimum: String,
+    },
+
+    #[error("{name} value {value} is less than or equal to {exclusive_minimum}")]
```

Comment:
hm - if we're going to show these to the user shouldn't they be more instructive? ie "Baseline must be greater than 10" instead of "Baseline value 12 is less than or equal to 10"